buildSrc/src/main/groovy/org/gradle/build/docs/UserGuideSectionVerifier.groovy

```diff
@@ -5,29 +5,34 @@ import groovy.transform.ToString
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.CacheableTask
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import org.w3c.dom.NodeList
 
 import javax.xml.parsers.DocumentBuilderFactory
 import javax.xml.xpath.XPathConstants
 import javax.xml.xpath.XPathFactory
 
-
 /**
  * This is a Gradle task that runs through the user guide's source XML files
  * and checks that all sections have {@code id} attributes and there are no
  * duplicate section IDs.
  */
 @CompileStatic
+@CacheableTask
 class UserGuideSectionVerifier extends DefaultTask {
     @InputFiles
     FileCollection docbookFiles
 
     @Input
     String fileEncoding = "UTF-8"
```

Comment:
File encoding is shown as "unused" in IDEA. Where is that required?

Reply:
Right. It's not used. Removed it.